Added Httplistener wrapper.

grokys opened the pull request:
This PR adds a wrapper for `HttpListener` which is notoriously difficult to mock.

I'm not sure if I've put everything in the right place, so please tell me if you'd like things to be moved!

FakeHaacked commented on src/Net/IHttpListenerRequest.cs:
```diff
@@ -0,0 +1,333 @@
+        /// A <see cref="string"/> array that contains the type names specified in the request's
+        /// Accept header or null if the client request did not include an Accept header.
+        /// </returns>
+        string[] AcceptTypes { get; }
```
For the most part, Rothko tries to follow the original API as closely as possible. But in cases like this where the API returns an array, I switch it to `IEnumerable<T>`. So in this specific case, this should be `IEnumerable<string>`

FakeHaacked commented on src/Net/IHttpListenerRequest.cs:
```diff
@@ -0,0 +1,333 @@
+        /// <see cref="HttpRequestHeader.AcceptLanguage"/> header or null if the client request
+        /// did not include an <see cref="HttpRequestHeader.AcceptLanguage"/> header.
+        /// </returns>
+        string[] UserLanguages { get; }
```
See other note about returning array types.

FakeHaacked commented on src/Net/HttpListenerResponseWrapper.cs:
```diff
@@ -0,0 +1,122 @@
+            Dispose(false);
+        }
+
+        void IDisposable.Dispose()
```
Is there a reason you're explicitly implementing this method? This should probably simply be `public void Dispose` and then you can remove the suppression.

grokys replied:
Just because in the original `HttpListener` it's (bizarrely) implemented like that. Now I know we don't have to follow the original API exactly, I'll make this look more normal.

Haacked commented:
Just a few notes. Once you fix these, you should be able to remove a lot of the Global Suppressions you added. I want to minimize the number of Code Analysis rules we suppress.

grokys commented:
@Haacked addressed your feedback. Also made `HttpListenerContextWrapper` non-sealed as we need to make it an import in ghfvs and inheriting from it is easier than wrapping the wrapper. 

I had to add a CA2213 suppression for `HttpListenerContextWrapper` because even though it does now dispose of the `Response`, CA doesn't seem to be able to detect this and warns anyway.

Haacked commented:
Thanks!
